.prettierrc.json

```diff
@@ -0,0 +1,19 @@
+{
+  "arrowParens": "always",
+  "bracketSameLine": false,
+  "bracketSpacing": true,
+  "embeddedLanguageFormatting": "off",
+  "htmlWhitespaceSensitivity": "css",
+  "insertPragma": false,
+  "jsxSingleQuote": false,
+  "printWidth": 80,
+  "proseWrap": "always",
+  "quoteProps": "as-needed",
+  "requirePragma": false,
+  "semi": true,
+  "singleAttributePerLine": false,
+  "singleQuote": false,
+  "tabWidth": 2,
+  "trailingComma": "es5",
+  "useTabs": false
+}
```

Comment:
Can you please explain what this file does?

Reply:
Sure! A `.prettierrc.json` file is a configuration file for the Prettier autoformatter. I previously set up the VSCode workspace to use this autoformatter. This is what GPT-4 had to say about each setting, but please note that this file was naively pasted from a different project, mainly for the `proseWrap` option. I should maybe have removed/edited the other settings. Let me know if you'd like me to do this.

> - `"arrowParens": "always"`: Include parentheses around a sole arrow function parameter.
> - `"bracketSameLine": false`: Puts the `>` of a multi-line JSX element at the end of the last line instead of being alone on the next line.
> - `"bracketSpacing": true`: Prints spaces between brackets in object literals.
> - `"embeddedLanguageFormatting": "off"`: Controls whether Prettier formats code that it embeds in the output (like code inside of markdown code blocks).
> - `"htmlWhitespaceSensitivity": "css"`: Specifies how whitespace in HTML files should be handled.
> - `"insertPragma": false`: Insert `@format` pragma into the top of files formatted by Prettier. This is usually to indicate that the file has been formatted with Prettier.
> - `"jsxSingleQuote": false"`: Use single quotes instead of double quotes in JSX.
> - `"printWidth": 80"`: Specifies the line length that the printer will wrap on.
> - `"proseWrap": "always"`: Wrap markdown text as if it were a prose.
> - `"quoteProps": "as-needed"`: Change when properties in objects are quoted. "as-needed" means only add quotes around object properties where required.
> - `"requirePragma": false"`: Prettier will only format files with a special comment, called a pragma, at the top of the file.
> - `"semi": true"`: Print semicolons at the ends of statements.
> - `"singleAttributePerLine": false"`: Enable or disable the printing of each JSX attribute on its own line when the attributes are spread over multiple lines.
> - `"singleQuote": false"`: Use single quotes instead of double quotes.
> - `"tabWidth": 2"`: Specifies the number of spaces per indentation level.
> - `"trailingComma": "es5"`: Print trailing commas wherever possible when multi-line. "es5" means trailing commas where valid in ES5 (objects, arrays, etc).
> - `"useTabs": false"`: Indent lines with tabs instead of spaces.